### PR TITLE
Issue/998 android

### PR DESCRIPTION
**Pull request by ichinaski**

### CODE REVIEW

File: GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java [hunk 1]
```diff
@@ -139,13 +139,8 @@ public SurveyInstance save(SurveyInstance si, DeviceFiles deviceFile) {
...
                     qas.getValue().lastIndexOf("/") + 1);
 
-            Device d = null;
-            if (deviceFile.getImei() != null) {
-                d = deviceDao.getByImei(deviceFile.getImei());
-            }
-            if (d == null && deviceFile.getPhoneNumber() != null) {
-                d = deviceDao.get(deviceFile.getPhoneNumber());
-            }
+            Device d = deviceDao.getDevice(deviceFile.getAndroidId(),
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/dao/SurveyInstanceDAO.java]: I noticed that this is within a loop. Perhaps you can shift it out and instead use a hash to store the device (if thats necessary at all).

File: GAE/src/org/waterforpeople/mapping/app/web/TaskServlet.java [hunk 2]
```diff
@@ -164,6 +165,7 @@ public TaskServlet() {
         deviceFile.setProcessDate(getNowDateTimeFormatted());
         deviceFile.setProcessedStatus(StatusCode.IN_PROGRESS);
         deviceFile.setURI(url);
+        deviceFile.setPhoneNumber(androidId);
```
Comment (muloem) [inline GAE/src/org/waterforpeople/mapping/app/web/TaskServlet.java]: `setAndroidId()`

**Comment by muloem** [comment]
Also perhaps we need to add the AndroidId to this set of params for the rescheduling.. https://github.com/akvo/akvo-flow/blob/ec09ea90ada7e21cc39181f8dbafb3e57c479d2c/GAE/src/org/waterforpeople/mapping/app/web/TaskServlet.java#L278-L303

File: GAE/src/org/waterforpeople/mapping/app/web/TaskServlet.java (hunk 2 above)
Comment (ichinaski) [inline GAE/src/org/waterforpeople/mapping/app/web/TaskServlet.java]: ouch!

**Comment by muloem** [comment]
So regarding the `/surveyedlocale` endpoint change, it was mainly because the whole block will possibly (even though highly unlikely) be skipped because of this.. 
https://github.com/akvo/akvo-flow/blob/0ec6905ff029b155c535f303b97906c6833ccca9/GAE/src/org/waterforpeople/mapping/app/web/SurveyedLocaleServlet.java#L83

**Comment by ichinaski** [comment]
The `DeviceSurveyJobQueueDAO` [catches breath again], already checks the androidId, IMEI, and phone number params. Perhaps we can fully remove this condition here?

**Comment by muloem** [comment]
Yep. sounds like a plan. clean away!

**Comment by muloem** [comment]
:+1: code reviewed!

**Pull request merged**